lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm

```diff
@@ -42,19 +44,65 @@ sub doInventory {
     # Get agent capabilities
     my @agentCapabilities = _listAgentCapabilities();
 
+    my $customProperties = _getCustomProperties(logger => $logger);
+
     my $rudder = {
         HOSTNAME => $hostname,
         UUID => $Uuid,
         AGENT => \@agents,
         SERVER_ROLES => { SERVER_ROLE => \@serverRoles },
         AGENT_CAPABILITIES => { AGENT_CAPABILITY => \@agentCapabilities },
+        CUSTOM_PROPERTIES => $customProperties,
     };
 
     $inventory->addEntry(
         section => 'RUDDER', entry => $rudder
     );
 }
 
+sub _getCustomProperties {
+    my (%params) = @_;
+    my $logger   = $params{logger};
+
+    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\local\hooks.d' : '/var/rudder/local/hooks.d';
+    my $custom_properties;
+    if (-d "$custom_properties_dir") {
+        my @custom_propertes_list = ();
+        opendir(DIR, $custom_properties_dir); # or die $!;
```

Comment:
You should remove the not useful comment at the end of line here.

Reply:
ha, i've had a remark on this one that I should catch the error. Should I catch, or will it be correctly handled by calling code ?

Reply:
To be fully safe, you can initialize @ordered_script_list to empty list, run an eval where you want to populate the list, you should even move the `closedir(DIR)` from l.99 in the eval. Then if something goes wrong, the list will be kept empty or even partially filled. If the list remains empty, the while will do nothing and no custom property will be inventoried, but the Rudder module will still finish and inventory other informations.